syft/distro/distro.go

```diff
@@ -20,9 +22,15 @@ func NewUnknownDistro() Distro {
 }
 
 func NewDistro(t Type, ver string) (Distro, error) {
-	verObj, err := hashiVer.NewVersion(ver)
-	if err != nil {
-		return Distro{}, fmt.Errorf("could not create distro version: %w", err)
+	var verObj *hashiVer.Version
```

Comment:
This isn't a required change at all, just a subjective suggestion for making the implementation easier to follow and slightly less nested (and consequently, it cuts down on variables that are declared without defined values). **Feel free to discard!** 😄 

Assuming we accept @wagoodman 's suggestion of using the zero value for `RawVersion` in this case (which I agree with), we can detect a lack of version early on and return a `Distro` accordingly.

```go
if ver == "" {
	return Distro{Type: t}, nil
}

verObj, err := hashiVer.NewVersion(ver)
if err != nil {
	return Distro{}, fmt.Errorf("could not create distro version: %w", err)
}

return Distro{
	Type:       t,
	Version:    verObj,
	RawVersion: ver,
}, nil
```

Reply:
great feedback! This is normally how I prefer to program things but I'm still getting my feet wet w/ go, so I missed this! It's funny how new contexts can be distracting this way. Updated to follow this format!